tests/unit/common/registers/uniqueness.cpp

```diff
@@ -0,0 +1,231 @@
+#include "registers/MachRegister.h"
+#include "Architecture.h"
+#include <iostream>
+#include <unordered_map>
+#include <string>
+#include <cstdlib>
+#include <map>
+#include <vector>
+#include <algorithm>
+
+/*
+ * Ensure all registers for an architecture have unique representations
+ */
+
+// Some architectures have known aliases, so exclude them.
+
+static std::vector<char const*> riscv64_aliases();
+
+using alias_map = std::map<Dyninst::Architecture, std::vector<char const*>>;
+alias_map known_aliases = {
+  {Dyninst::Arch_aarch64,
+    {
+      "aarch64::x29", "aarch64::fp",
+      "aarch64::x30", "aarch64::lr",
+      "aarch64::Ip0", "aarch64::x16",
+      "aarch64::Ip1", "aarch64::x17",
+    }
+  },
+  {Dyninst::Arch_x86, {}},
+  {Dyninst::Arch_x86_64, {}},
+  {Dyninst::Arch_ppc64, {}},
+  {Dyninst::Arch_amdgpu_gfx908, {}},
+  {Dyninst::Arch_amdgpu_gfx90a, {}},
+  {Dyninst::Arch_amdgpu_gfx940, {}},
+  {Dyninst::Arch_riscv64, { riscv64_aliases() }},
+};
+
+bool check(Dyninst::Architecture arch) {
+  auto const& regs = Dyninst::MachRegister::getAllRegistersForArch(arch);
+
+  auto known_alias = [arch](std::string const& name) {
+    auto const& aliases = known_aliases[arch];
+    return std::find(aliases.begin(), aliases.end(), name) != aliases.end();
+  };
+
+  auto success = true;
+
+  std::unordered_map<unsigned int, std::string> vals;
+  int i = 0;
+  for(auto r : regs) {
+    auto res = vals.insert({r.val(), r.name()});
+    auto itr = res.first;
+    auto inserted = res.second;
+    if(!inserted && !known_alias(r.name())) {
+      auto val = itr->first;
+      auto name = itr->second;
+      std::cerr << "alias: 0x" << std::hex << val
+                << "  " << r.name() << "  " << std::dec << i << "\n";
```

Comment:
Just wandering, what information does this i variable provides?

Reply:
Oops. That's some debug code. I'll remove it.